main/integration-tests/__test__/api-tests/user/update-user.test.js

```diff
@@ -0,0 +1,103 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const { runSetup } = require('../../../support/setup');
+const errorCode = require('../../../support/utils/error-code');
+
+describe('Update current user scenarios', () => {
+  let setup;
+
+  beforeAll(async () => {
+    setup = await runSetup();
+  });
+
+  afterAll(async () => {
+    await setup.cleanup();
+  });
+
+  describe('updating current user', () => {
+    it('should fail for anonymous user', async () => {
+      const anonymousSession = await setup.createAnonymousSession();
+      await expect(anonymousSession.resources.currentUser.update({ rev: 0 })).rejects.toMatchObject({
+        code: errorCode.http.code.badImplementation,
+      });
+    });
+
+    it('should update current user regardless of the uid in request body', async () => {
+      const researcher1Session = await setup.createResearcherSession();
+      const researcher2Session = await setup.createResearcherSession();
+      const researcher2Info = await researcher2Session.resources.currentUser.get();
+      const researcher1Info = await researcher1Session.resources.currentUser.get();
+      await expect(
+        researcher1Session.resources.currentUser.update({ uid: researcher2Info.uid, rev: 0, status: 'pending' }),
+      ).resolves.toEqual(expect.objectContaining({ uid: researcher1Info.uid }));
+    });
+
+    it('should fail if non-admin user update restrictive fields', async () => {
```

Comment:
Maybe we can use it.each() this way you can provide a list of all the restricted fields and the test will run for each on of them.